sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/KeyvaultKeyTest.groovy

```diff
@@ -0,0 +1,155 @@
+package com.azure.storage.blob.specialized.cryptography
+
+import com.azure.core.credential.TokenCredential
+import com.azure.core.cryptography.AsyncKeyEncryptionKey
+import com.azure.core.http.HttpClient
+import com.azure.core.http.HttpPipeline
+import com.azure.core.http.HttpPipelineBuilder
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy
+import com.azure.core.http.policy.ExponentialBackoff
+import com.azure.core.http.policy.HttpLogDetailLevel
+import com.azure.core.http.policy.HttpLogOptions
+import com.azure.core.http.policy.HttpLoggingPolicy
+import com.azure.core.http.policy.HttpPipelinePolicy
+import com.azure.core.http.policy.HttpPolicyProviders
+import com.azure.core.http.policy.RetryPolicy
+import com.azure.core.http.policy.RetryStrategy
+import com.azure.core.http.policy.UserAgentPolicy
+import com.azure.core.test.TestMode
+import com.azure.core.util.Configuration
+import com.azure.identity.ClientSecretCredentialBuilder
+import com.azure.identity.DefaultAzureCredentialBuilder
+import com.azure.security.keyvault.keys.KeyClient
+import com.azure.security.keyvault.keys.KeyClientBuilder
+import com.azure.security.keyvault.keys.KeyServiceVersion
+import com.azure.security.keyvault.keys.cryptography.KeyEncryptionKeyClientBuilder
+import com.azure.security.keyvault.keys.models.CreateRsaKeyOptions
+import com.azure.security.keyvault.keys.models.KeyVaultKey
+import com.azure.storage.blob.BlobContainerClient
+import com.azure.storage.common.implementation.Constants
+
+import java.time.Duration
+import java.time.OffsetDateTime
+
+class KeyvaultKeyTest extends APISpec {
+
+    BlobContainerClient cc
+    EncryptedBlobClient bec // encrypted client for download
+    KeyClient keyClient
+    String keyId
+
+    def setup() {
+        def keyVaultUrl = "https://azstoragesdkvault.vault.azure.net/"
+        if (testMode != TestMode.PLAYBACK) {
+            keyVaultUrl = Configuration.getGlobalConfiguration().get("KEYVAULT_URL")
+        }
+
+        KeyClientBuilder builder = new KeyClientBuilder()
+            .pipeline(getHttpPipeline(getHttpClient(), KeyServiceVersion.V7_0))
+
+        keyClient = builder
+            .httpClient(getHttpClient())
+            .vaultUrl(keyVaultUrl)
+            .buildClient()
+
+        keyId = generateResourceName("keyId", entityNo++)
+
+        KeyVaultKey keyVaultKey = keyClient.createRsaKey(new CreateRsaKeyOptions(keyId)
+            .setExpiresOn(OffsetDateTime.now().plusYears(1))
+            .setKeySize(2048))
+
+        AsyncKeyEncryptionKey akek = new KeyEncryptionKeyClientBuilder()
+            .credential(new DefaultAzureCredentialBuilder().build())
```

Comment:
Ideally, I would build a pipeline here and in playback mode skip adding the Bearertoken policy in the pipeline.
Then set the pipeline on the builder.

But, to keep the setup as it is, this might work too. The credential should return a dummy token in playback mode.
 .credential(trc -> Mono.just(new AccessToken("Dummy-Token", OffsetDateTime.now().plusHours(2))));

Reply:
Thanks for catching that! Forgot about this client